crates/zkvm/adapters/native/src/input.rs

```diff
@@ -0,0 +1,37 @@
+use strata_zkvm::{ZkVmInputBuilder, ZkVmResult};
+
+use crate::zkvm::NativeMachine;
+
+pub struct NativeMachineInputBuilder(pub NativeMachine);
+
+impl<'a> ZkVmInputBuilder<'a> for NativeMachineInputBuilder {
+    type Input = NativeMachine;
+
+    fn new() -> NativeMachineInputBuilder {
+        Self(NativeMachine::new())
+    }
+
+    fn write_buf(&mut self, item: &[u8]) -> ZkVmResult<&mut Self> {
+        self.0.write_slice(item.to_vec());
+        Ok(self)
+    }
+
+    fn write_serde<T: serde::Serialize>(&mut self, item: &T) -> ZkVmResult<&mut Self> {
+        let slice = bincode::serialize(&item)?;
+        self.write_buf(&slice)
+    }
+
+    fn write_borsh<T: borsh::BorshSerialize>(&mut self, item: &T) -> ZkVmResult<&mut Self> {
+        let slice = borsh::to_vec(item)?;
+        self.write_buf(&slice)
+    }
+
+    fn write_proof(&mut self, item: strata_zkvm::AggregationInput) -> ZkVmResult<&mut Self> {
+        // TODO: figure this out
+        self.write_buf(item.proof().as_bytes())
+    }
+
+    fn build(&mut self) -> ZkVmResult<Self::Input> {
+        Ok(self.0.clone())
+    }
```

Comment:
This is relatively minor but since we clone in `.build` and *also* make a copy in `.write_buf` then we have to make *two* copies of any buffer we pass to `.write_buf`.  Ideally in the builder we could just store a ref to the buf passed, but I can see that this interferes with how the other write fns here work because they only own the copy in the scope of the function.  Not sure the best way to proceed with that.  At the very least you could require passing a `Vec<u8>` to try to avoid the extra clone.  Maybe similar with adding a `.into_bytes()` on `AggregationInput`?

This probably doesn't make any measurable performance difference since the only place it even *might* matter is the L1 proofs, which we only do once every 10 mins.

Reply:
Not sure on how to proceed. Leaving it as is for now.